hacspec/src/poly.rs

```diff
@@ -267,38 +267,118 @@ pub(crate) fn extended_euclid_invert<T: TRestrictions<T>>(x: T, n: T, signed: bo
     t
 }
 
-/// Extended euclidean algorithm to compute the inverse of x in yℤ[x]
-#[inline]
-pub fn extended_euclid<T: TRestrictions<T>>(x: &[T], y: &[T], n: T) -> Result<Vec<T>, &'static str> {
-    let (x, y) = normalize(x, y);
-    println!("euclid: {:?} / {:?}", x, y);
-
-    let mut new_t = vec![T::default(); x.len()];
-    new_t[0] = T::from_literal(1);
+// Pad to degree d and reverse coefficients
+fn reverse<T: TRestrictions<T>>(x: &[T], d: usize) -> Vec<T> {
+    let mut out = vec![T::default(); d+1];
+    for i in 0..std::cmp::min(x.len(), d+1) {
+        out[d-i] = x[i]; //XXX: this could probably be written more nicely with iterators?
+    }
+    out 
+}
 
-    let mut new_r = x.clone();
+/// Subtract quotient (bn/x^bd) from (an/x^ad)
+fn quot_sub<T: TRestrictions<T>>(an: &[T], ad: usize, bn: &[T], bd: usize, n: T) -> (Vec<T>, usize){
+    let cd = std::cmp::max(ad, bd);
+    let x = monomial(T::from_literal(1),1);
+    let mut a = an.to_vec();
+    let mut b = bn.to_vec();
+    for _ in 0..cd-ad {
```

Comment:
Hm, I don't like these loops but I also don't have a good idea how to make them nicer ... (maybe just ignore this comment :))